mindsdb/integrations/libs/ml_exec_base.py

```diff
@@ -369,6 +369,9 @@ def predict(self, model_name: str, data: list, pred_format: str = 'dict',
             args['dtype_dict'] = predictor_record.dtype_dict
             args['learn_args'] = predictor_record.learn_args
 
+        if self.handler_class.__name__ in ('LangChainHandler',):
```

Comment:
Instead I would put this code into _get_ml_handler
```
    command_executor = None
    if it is LangChainHandler:
         sql_session = SessionController()
         command_executor = ExecuteCommands(sql_session, None)
    ml_handler = self.handler_class(
                engine_storage=handlerStorage,
                model_storage=modelStorage,
                executor=command_executor,
     )
```

Reply:
That would require changing the signature of this method to return both the handler and its arguments, right? I'm not sure if that is desirable, perhaps an `_override_handler_args()` method would be better instead?

Reply:
Do you mean signature of handler_class? If so it is encapsulated from developer and no need to change ml handlers

Reply:
Indeed, that's the one I meant. My concern is mostly semantics: if the method is called `handler_class`, isn't it confusing that it returns handler arguments as well?

I see above this code snippet there is similar logic for the lightwood handler, so my proposal is to take both of these and move them into a new method that would solely focus on overriding certain handler arguments depending on their name/type.